fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/FhirSchemaGenerator.java

```diff
@@ -544,6 +594,140 @@ public void addLogicalResources(PhysicalDataModel pdm) {
         pdm.addObject(tbl);
     }
 
+    /**
+     * Create the COMMON_CANONICAL_VALUES table. Used from schema V0014 to normalize
+     * meta.profile search parameters (similar to common_token_values). Only the url
+     * is included by design. The (optional) version and fragment values are stored
+     * in the parameter mapping table (logical_resource_profiles) in order to support
+     * inequalities on version while still using a literal CANONICAL_ID = x predicate.
+     * These canonical ids are cached in the server, so search queries won't need to
+     * join to this table. The URL is typically a long string, so by normalizing and
+     * storing/indexing it once, we reduce space consumption.
+     * @param pdm
+     */
+    public void addCommonCanonicalValues(PhysicalDataModel pdm) {
+        final String tableName = COMMON_CANONICAL_VALUES;
+        final String unqCanonicalUrl = "UNQ_" + tableName + "_URL";
+        Table tbl = Table.builder(schemaName, tableName)
+                .setVersion(FhirSchemaVersion.V0014.vid())
+                .setTenantColumnName(MT_ID)
+                .addBigIntColumn(CANONICAL_ID, false)
+                .addVarcharColumn(URL, CANONICAL_URL_BYTES, false)
+                .addPrimaryKey(tableName + "_PK", CANONICAL_ID)
+                .addUniqueIndex(unqCanonicalUrl, URL)
+                .setTablespace(fhirTablespace)
+                .addPrivileges(resourceTablePrivileges)
+                .enableAccessControl(this.sessionVariable)
+                .build(pdm);
+
+        // TODO should not need to add as a table and an object. Get the table to add itself?
```

Comment:
What is needed to resolve the TODO?

Reply:
Needs some refactor/cleanup of the framework - issue #2557